### PR TITLE
chore(#27): improve dev experience with more docs and scripts

### LINKED ISSUE #27: Discussion: branches as main release vehicle
Now that we have cut a few releases with the new process, I wanted to re-visit our strategy here to add some additional pros/cons to the previous discussion.  

Currently, our [release process](https://github.com/medic/cht-monitoring/tree/main/development#developing) involves creating a new branch for every release (every commit to `main`) and then instructing consumers to checkout particular branches locally to get the changes from the release. Tags are also created as part of the process, but are not used.

## Pain points

1. Does not conform to (my understanding) of the git paradigm for branches/tags. Typically when interacting with a git repo, you would view tags as pointer to a fixed commit that will never change and branches, on the other hand, refer to a stream of changes that may include additions in the future.  Treating branches as release pointers will be surprising behavior for anyone with git experience.
2. Does not work well with tooling. GitHub releases assumes that every release is associated with a tag.  If you look at the [release page](https://github.com/medic/cht-monitoring/releases/tag/1.1.0) for a release, you will see info about the tag and commit hash, but not the branch.  Additionally, tools we may want to use in the future, like `semantic-release`, are going to be more difficult to script/configure to operate within a non-standard paradigm
3. "Fix" releases end up with non-standard (in semantic versioning terms) branch names like `1.1.1.x` with the extra `.x` on the end.  I am afraid this will just cause confusion for folks that are familiar with semantic versioning.  

## Proposal

I think we should stick with the same branching pattern as cht-core and just create branches for the major and minor releases.  Fix commits should be just added to the latest existing branch release (e.g. `1.1.x` currently) and should not have a _new_ branch associated with them.  Our deployment documentation can then be updated to note that checking out a new cht-monitoring branch is how to _get new features_, but just doing a `git pull` from the checked out branch is how to get bug fixes.  This will go a long way towards resolving `#1` and `#3`!

Additionally, I know that the main reason we decided to recommend admins checkout branches instead of tags is to avoid the `detached HEAD` warning.  A different way to avoid the warning would be to just recommend checking out tags with the `advice.detachedHead=false` flag set. E.g: `git -c advice.detachedHead=false checkout 1.1.0`.  This will suppress the warning that is typically displayed in the terminal.  This approach would mean that consumers could just target the tags directly (and we would not have to document the separate `git pull` process mentioned above for fixes).  

Anyway, these are just my thoughts. Hoping for an open and productive discussion here!  @mrjones-plip @m5r @eljhkrr

### PR DESCRIPTION
**Pull request by jkuester**
The goal here is to continue improving the development experience with cht-monitoring.

Changes:
- Add additional development documentation with pointers on where to get started for adding/modifying Prometheus and Grafana resources
- Add development documentation for including additional Prometheus exporters
- Add npm script to make it easier to patch the scrape interval configuration when doing development/testing
- Make linting a top-level NPM script (instead of just fake-cht) so that all the development scripts can be linted too!

Closes #27

### CODE REVIEW

File: development/README.md [hunk 1]
```diff
@@ -8,11 +10,76 @@ This repo has a manual release process where each feature/bug fix will be releas
 3. Have the PR reviewed
 4. Merge the PR to `main`
 5. Following SemVer create either a `N.X.X` (eg Major `1.x.x`), `N.N.X` (eg Minor `1.1.x`) or `N.N.N.X` (eg Patch `1.1.1.x`) branch.  Push this branch to repo.
-6. Create a [new release](https://github.com/medic/cht-monitoring/releases/new) being sure to specify the Target branch (eg `1.1.x`) and set the correct release (eg `1.1.0`) and release notes.  Here's an example of releasing a fictitious `1.1.0` release:
-
-   ![Screenshot showing GitHub "Create new Release" page in this repo](./release.png)
```
Comment (jkuester) [inline development/README.md:13]: This screenshot does not exist...

Comment (mrjones-plip) [reply development/README.md:13]: fixed in 1892723

**Review by mrjones-plip — APPROVED** [review]
This is SO great - thank you @jkuester ! It will be great to share with other teammates who are looking to start adding to the repo.

I saw you added @m5r to the PR - perfect! It'll be great to get @m5r's feedback as he's new to this and we want to ensure it's easy for a new comer to follow :dart:

**Comment by mrjones-plip** [comment]
@jkuester  - fixed the missing image - if it's too chonky feel free to yank it back out!

**Comment by jkuester** [comment]
Oh, hah!  Thanks @mrjones-plip for the screenshot fix!  I figured that maybe got lost when moving docs around and did not want to bother you about it since we can hopefully get https://github.com/medic/cht-monitoring/issues/34 done soon.  I did not see that the screenshot actually still existed in the repo...  :facepalm:

**Review by m5r — APPROVED** [review]
Good changes all around!

TIL `patch` is preinstalled with many distros and easily installable on Windows with [GnuWin](http://gnuwin32.sourceforge.net/)

**Pull request merged**